### PR TITLE
Impose limit on context size

### PR DESCRIPTION
**Pull request by edsko**

### CODE REVIEW

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs [hunk 1]
```diff
@@ -28,27 +28,27 @@ module Cardano.Prelude.GHC.Heap.NormalForm.Classy (
...
+
+import qualified Data.ByteString                     as BS.Strict
+import qualified Data.ByteString.Lazy                as BS.Lazy
+import qualified Data.IntMap                         as IntMap
+import qualified Data.Map                            as Map
+import qualified Data.Set                            as Set
+import qualified Data.Text                           as Text.Strict
+import qualified Data.Text.Lazy                      as Text.Lazy
+import qualified Data.Vector                         as Vector.Boxed
+import qualified Data.Vector.Unboxed                 as Vector.Unboxed
```
Comment (intricate) [inline src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: Should probably revert the formatting here :smile:

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs [hunk 2]
```diff
@@ -84,20 +84,33 @@ class NoUnexpectedThunks a where
...
+          }
     where
       ctxt' :: [String]
       ctxt' = showTypeOf (Proxy @a) : ctxt
 
+      -- If we exceed the maximum context length, it is likely that there is a
+      -- loop in a 'NoUnexpectedContext' instance somewhere; it is unlikely that
+      -- a type is so complicated it exceeds the maximum context length.
+      maxContextLength :: Int
+      maxContextLength = 1000
```
Comment (intricate) [inline src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: Should the `maxContextLength` instead be specified by the caller?

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs (hunk 1 above)
Comment (edsko) [reply src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: Agreed.

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs (hunk 2 above)
Comment (edsko) [reply src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: I guess that's not unreasonable.

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs (hunk 1 above)
Comment (edsko) [reply src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: Oh, you did already. Good man :)

File: src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs (hunk 2 above)
Comment (intricate) [reply src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs]: I think this is okay for now but just a thought.

**Comment by nc6** [comment]
Totally dead; the `NoThunks` stuff has now moved to its own package. Closing!

**Pull request closed without merging**